src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,142 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(last,value,null);
+        if (first == null) {
+            first = last = newNode;
+        } else {
+            last.next = newNode;
+            last.next.prev = last;
+            last = last.next;
+        }
+        last = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        Objects.checkIndex(index,size + 1);
+        if (first == null) {
+            Node<T> newNode = new Node<>(first,value,last);
+            first = last = newNode;
+        } else if (index == 0) {
+            Node<T> newNode = new Node<>(null,value,first);
+            newNode.next = first;
+            first = newNode;
+        } else if (index == size) {
+            Node<T> newNode = new Node<>(last,value,null);
+            newNode.prev = last;
+            last.next = newNode;
+            last = newNode;
+        } else {
+            Node<T> nodeAtIndex = getNodeByIndex(index);
+            Node<T> newNode = new Node<>(nodeAtIndex.prev,value,nodeAtIndex);
+            newNode.next = nodeAtIndex;
+            newNode.prev = nodeAtIndex.prev;
+            if (newNode.prev != null) {
+                newNode.prev.next = newNode;
+            }
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (int i = 0; i <= list.size() - 1; i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Objects.checkIndex(index,size);
+        return getNodeByIndex(index).element;
```

Comment:
You always call `checkIndex `before `getNodeByIndex`. This is code repeating. You should simply call `checkIndex` first line inside `getNodeByIndex`. 
![image](https://user-images.githubusercontent.com/20326858/236793974-dab483f7-d4d1-402d-b993-c69e871fe73b.png)